### PR TITLE
Ops 3126 add test loadtest 01

### PR DESCRIPTION
**Pull request by RalfNellesCap**
# Description

## Approval for review

- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done

More and detailed information on the _definition of done_ can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

**Review by mamutmk5 — CHANGES_REQUESTED** [review]

File: ansible/playbook_rollout.yml [hunk 1]
```diff
@@ -38,3 +38,7 @@
     - role: antivirus_check_service
     - role: rocketchat
     - role: ingress
+    - role: dof_mongo
+      when: WITH_DATABASES is defined and WITH_DATABASES|bool
+    - role: dof_postgresql
+      when: WITH_DATABASES is defined and WITH_DATABASES|bool
```
[the remark this answers is not in the document]
Comment (RalfNellesCap) [reply ansible/playbook_rollout.yml]: added to group_vars/all group_vars/develop/
used condition and state to rollout and remove DBs

File: ansible/roles/dof_mongo/tasks/main.yml [hunk 2]
```diff
@@ -1,24 +1,28 @@
...
     kubernetes.core.k8s:
+      state: "{{ 'present' if WITH_DATABASES else 'absent' }}"
       kubeconfig: ~/.kube/config
       namespace: "{{ NAMESPACE }}"
       template: svc.yml.j2
       
-  - name: Persistent Volumes Claim  
+  - name: Persistent Volumes Claim
     kubernetes.core.k8s:
+      state: "{{ 'present' if WITH_DATABASES else 'absent' }}"
```
Comment (mamutmk5) [inline ansible/roles/dof_mongo/tasks/main.yml:10]: do not work so.
What is with with vars in the templates who are not set

File: ansible/roles/dof_mongo/tasks/main.yml [hunk 3]
```diff
@@ -1,24 +1,28 @@
...
       kubeconfig: ~/.kube/config 
       namespace: "{{ NAMESPACE }}"
       template: pvc.yml.j2
 
   - name: Configmap
     kubernetes.core.k8s:
+      state: "{{ 'present' if WITH_DATABASES else 'absent' }}"
```
Comment (mamutmk5) [inline ansible/roles/dof_mongo/tasks/main.yml:17]: do not work so.
What is with with vars in the templates who are not set

File: ansible/roles/dof_mongo/tasks/main.yml [hunk 4]
```diff
@@ -1,24 +1,28 @@
...
       kubeconfig: ~/.kube/config
       namespace: "{{ NAMESPACE }}"
       template: configmap.yml.j2
       apply: yes
 
   - name: Deployment
     kubernetes.core.k8s:
+      state: "{{ 'present' if WITH_DATABASES else 'absent' }}"
```
Comment (mamutmk5) [inline ansible/roles/dof_mongo/tasks/main.yml:25]: do not work so.
What is with with vars in the templates who are not set

File: ansible/roles/dof_postgresql/tasks/main.yml [hunk 5]
```diff
@@ -1,17 +1,20 @@
   - name: Service
     kubernetes.core.k8s:
+      state: "{{ 'present' if WITH_DATABASES else 'absent' }}"
```
Comment (mamutmk5) [inline ansible/roles/dof_postgresql/tasks/main.yml:3]: do not work so.
What is with with vars in the templates who are not set

File: ansible/roles/dof_postgresql/tasks/main.yml [hunk 6]
```diff
@@ -1,17 +1,20 @@
...
       kubeconfig: ~/.kube/config 
       namespace: "{{ NAMESPACE }}"
       template: svc.yml.j2
       
   - name: Persistent Volumes Claim  
     kubernetes.core.k8s:
+      state: "{{ 'present' if WITH_DATABASES else 'absent' }}"
```
Comment (mamutmk5) [inline ansible/roles/dof_postgresql/tasks/main.yml:10]: do not work so.
What is with with vars in the templates who are not set

File: ansible/roles/dof_postgresql/tasks/main.yml [hunk 7]
```diff
@@ -1,17 +1,20 @@
...
       kubeconfig: ~/.kube/config 
       namespace: "{{ NAMESPACE }}"
       template: pvc.yml.j2
       
   - name: Configmap
     kubernetes.core.k8s:
+      state: "{{ 'present' if WITH_DATABASES else 'absent' }}"
```
Comment (mamutmk5) [inline ansible/roles/dof_postgresql/tasks/main.yml:17]: do not work so.
What is with with vars in the templates who are not set

File: ansible/roles/dof_postgresql/tasks/main.yml [hunk 8]
```diff
@@ -20,21 +23,24 @@
       
   - name: Secret
     kubernetes.core.k8s:
-      kubeconfig: ~/.kube/config 
+      state: "{{ 'present' if WITH_DATABASES else 'absent' }}"
```
Comment (mamutmk5) [inline ansible/roles/dof_postgresql/tasks/main.yml:26]: do not work so.
What is with with vars in the templates who are not set

File: ansible/roles/dof_postgresql/tasks/main.yml [hunk 9]
```diff
@@ -20,21 +23,24 @@
...
+      kubeconfig: ~/.kube/config
       namespace: "{{ NAMESPACE }}"
       template: secret.yml.j2
       apply: yes
     when: ONEPASSWORD_OPERATOR is undefined or ONEPASSWORD_OPERATOR is defined and not ONEPASSWORD_OPERATOR
       
   - name: Secred by 1Password
     kubernetes.core.k8s:
+      state: "{{ 'present' if WITH_DATABASES else 'absent' }}"
```
Comment (mamutmk5) [inline ansible/roles/dof_postgresql/tasks/main.yml:35]: do not work so.
What is with with vars in the templates who are not set

File: ansible/roles/dof_postgresql/tasks/main.yml [hunk 10]
```diff
@@ -20,21 +23,24 @@
...
       kubeconfig: ~/.kube/config 
       namespace: "{{ NAMESPACE }}"
       template: onepassword.yml.j2
     when: ONEPASSWORD_OPERATOR is defined and ONEPASSWORD_OPERATOR|bool
       
   - name: Deployment
     kubernetes.core.k8s:
+      state: "{{ 'present' if WITH_DATABASES else 'absent' }}"
```
Comment (mamutmk5) [inline ansible/roles/dof_postgresql/tasks/main.yml:43]: do not work so.
What is with with vars in the templates who are not set

File: ansible/host_vars/test-loadtest-01/with.yml [hunk 11]
```diff
@@ -0,0 +1,2 @@
+WITH_STORAGE: true
+# WITH_MONGODB??
```
Comment (mamutmk5) [inline ansible/host_vars/test-loadtest-01/with.yml:2]: delete this line

**Review by mamutmk5 — CHANGES_REQUESTED** [review]

File: ansible/roles/dof_postgresql/tasks/main.yml (hunk 10 above)
Comment (RalfNellesCap) [reply ansible/roles/dof_postgresql/tasks/main.yml:43]: Simon suggested to use default values instead. done

File: ansible/roles/dof_mongo/templates/deployment.yml.j2 [hunk 12]
```diff
@@ -29,7 +29,7 @@ spec:
         runAsNonRoot: true
       containers:
       - name: mongo
-        image: {{ MONGO_IMAGE_NAME }}:{{ MONGO_IMAGE_TAG }}
+        image: {{ MONGO_IMAGE_NAME|default("mongo", true) }}:{{ MONGO_IMAGE_TAG|default("4.2.9", true) }}
```
Comment (mamutmk5) [inline ansible/roles/dof_mongo/templates/deployment.yml.j2:32]: No, not default set her

File: ansible/roles/dof_postgresql/templates/configmap.yml.j2 [hunk 13]
```diff
@@ -7,7 +7,7 @@ metadata:
     app: postgres
 data:
   # general
-  POSTGRES_DB: "{{ POSTGRES_DB }}"
-  POSTGRES_USER: "{{ POSTGRES_USER }}"
+  POSTGRES_DB: "{{ POSTGRES_DB|default("calendar", true) }}"
+  POSTGRES_USER: "{{ POSTGRES_USER|default("node", true) }}"
```
Comment (mamutmk5) [inline ansible/roles/dof_postgresql/templates/configmap.yml.j2:11]: No, not default set her

File: ansible/roles/dof_postgresql/templates/deployment.yml.j2 [hunk 14]
```diff
@@ -19,7 +19,7 @@ spec:
     spec:
       containers:
       - name: postgres
-        image: {{ POSTGRES_IMAGE_NAME }}:{{ POSTGRES_IMAGE_TAG }}
+        image: {{ POSTGRES_IMAGE_NAME|default("postgres", true) }}:{{ POSTGRES_IMAGE_TAG|default("11.9", true) }}
```
Comment (mamutmk5) [inline ansible/roles/dof_postgresql/templates/deployment.yml.j2:22]: No, not default set her

File: ansible/roles/dof_postgresql/templates/secret.yml.j2 [hunk 15]
```diff
@@ -8,4 +8,4 @@ metadata:
   namespace: {{ NAMESPACE }}
 type: Opaque
 data:
-  POSTGRES_PASSWORD: "{{ POSTGRES_PASSWORD | b64encode }}"
\ No newline at end of file
+  POSTGRES_PASSWORD: "{{ POSTGRES_PASSWORD|default("postgres", true) | b64encode }}"
```
Comment (mamutmk5) [inline ansible/roles/dof_postgresql/templates/secret.yml.j2:11]: No, not default set her

**Review by mamutmk5 — CHANGES_REQUESTED** [review]

**Review by mamutmk5 — DISMISSED** [review]

**Review by simoncolincap — APPROVED** [review]

**Review by mamutmk5 — APPROVED** [review]

**Pull request merged**